From 18f5502d11dda83a25d3f04870f1e08d96901a45 Mon Sep 17 00:00:00 2001
From: Volker Theile <votdev@gmx.de>
Date: Tue, 13 Jul 2021 10:07:52 +0200
Subject: [PATCH] Failed to load /var/cache/salt/minion/roots/mtime_map when it
 contains invalid UTF-8 data.

UTF-8 encoding was forced by https://github.com/saltstack/salt/commit/5ea3fb2cf241bbc18656fc65752f0a92e950266d, but decoding exceptions are not handled. This will be fixed by this PR.

Fixes: https://github.com/saltstack/salt/issues/60535

Signed-off-by: Volker Theile <votdev@gmx.de>
---
 salt/fileserver/roots.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/salt/fileserver/roots.py b/salt/fileserver/roots.py
index 54dda2d9ee5dc..f0fefc9fe2919 100644
--- a/salt/fileserver/roots.py
+++ b/salt/fileserver/roots.py
@@ -172,7 +172,7 @@ def update():
                         mtime_map_path,
                         line,
                     )
-    except OSError:
+    except (OSError, UnicodeDecodeError):
         pass

     # compare the maps, set changed to the return value
@@ -266,6 +266,7 @@ def file_hash(load, fnd):
         except (
             os.error,
             OSError,
+            UnicodeDecodeError,
         ):  # Can't use Python select() because we need Windows support
             log.debug("Fileserver encountered lock when reading cache file. Retrying.")
             # Delete the file since its incomplete (either corrupted or incomplete)
